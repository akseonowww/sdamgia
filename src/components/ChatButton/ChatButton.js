import React from 'react';

import './ChatButton.scss';

const ChatButton = () => {
	return <div className="ChatButton">Чат</div>;
};

export default ChatButton;
