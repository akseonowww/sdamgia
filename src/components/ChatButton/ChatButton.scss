.ChatButton {
	position: fixed;
	bottom: 50px;
	left: 15px;

	width: 30px;
	padding: 5px;
	height: 15px;

	font-size: 10pt;

	background: #f0f0f0;
	border: 1px solid #000;
	border-radius: 5px;
	cursor: pointer;
}
