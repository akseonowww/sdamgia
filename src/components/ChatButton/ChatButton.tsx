import React, { FC } from 'react'
import { cn } from '@bem-react/classname'
import { classnames as cx } from '@bem-react/classnames'

import './ChatButton.scss'

interface IChatButtonProps {}

const ChatButton = ({}: FC<IChatButtonProps>) => {
  const chatButton = cn('ChatButton')

  return <div className={cx(chatButton())}>Чат</div>
}

export default ChatButton
