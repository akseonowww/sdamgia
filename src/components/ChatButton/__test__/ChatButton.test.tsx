import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import ChatButton from '../ChatButton'

describe('ChatButton', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders without crashing', () => {
    act(() => {
      render(<ChatButton />, container)
    })
    expect(container && container.textContent).toBe('Чат')
  })

  it('should render a content', () => {
    act(() => {
      render(<ChatButton />, container)
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
