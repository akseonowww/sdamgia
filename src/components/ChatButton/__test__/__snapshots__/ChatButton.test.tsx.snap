// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatButton should render a content 1`] = `"<div class=\\"ChatButton\\">Чат</div>"`;
